include/aca_grpc.h

```diff
@@ -21,55 +21,88 @@
 
 using namespace alcor::schema;
 using grpc::Server;
+using grpc::ServerAsyncReader;
+using grpc::ServerAsyncReaderWriter;
+using grpc::ServerAsyncResponseWriter;
 using grpc::ServerBuilder;
+using grpc::ServerCompletionQueue;
 using grpc::ServerContext;
 using grpc::ServerReader;
 using grpc::ServerReaderWriter;
-using grpc::ServerAsyncResponseWriter;
-using grpc::ServerAsyncReader;
-using grpc::ServerAsyncReaderWriter;
-using grpc::ServerCompletionQueue;
 using grpc::ServerWriter;
 using grpc::Status;
 
 class GoalStateProvisionerAsyncServer {
   public:
+  /*
+    Base class that represents a gRPC call.
+    When you have a new kind of rpc, add the corresponding enum to CallType
+  */
+  struct AsyncGoalStateProvionerCallBase {
+    enum CallType { PUSH_NETWORK_RESOURCE_STATES, PUSH_GOAL_STATE_STREAM };
```

Comment:
Very good to have a small enum like this. What is the difference between the two? A short comment would help.

Reply:
Added